torch/csrc/utils/python_arg_parser.h

```diff
@@ -647,16 +660,7 @@ inline std::vector<int64_t> PythonArgs::intlistWithDefault(
         throw_intlist_exception(this, i, obj, idx);
       }
     } else {
-      // convert tensor to scalar outside of try / catch,
-      // so that Tensor subclass exceptions will not be caught.
-      if (THPUtils_checkLongExact(obj)) {
```

Comment:
Removing this is NOT ok. This is an important perf optimization by avoiding an expensive `isinstance()` check for basic object.

Reply:
sorry, i got it now.

And there should be a duplicate check here, so replaced `THPUtils_unpackIndex` with `THPUtils_unpackLong`

https://github.com/pytorch/pytorch/blob/431d25a141105c4037ad72017ad2d4da668c490a/torch/csrc/utils/python_arg_parser.h#L655

https://github.com/pytorch/pytorch/blob/431d25a141105c4037ad72017ad2d4da668c490a/torch/csrc/utils/python_numbers.h#L106